src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -1,34 +1,77 @@
-export const TodoFilter = () => (
-  <form className="field has-addons">
-    <p className="control">
-      <span className="select">
-        <select data-cy="statusSelect">
-          <option value="all">All</option>
-          <option value="active">Active</option>
-          <option value="completed">Completed</option>
-        </select>
-      </span>
-    </p>
-
-    <p className="control is-expanded has-icons-left has-icons-right">
-      <input
-        data-cy="searchInput"
-        type="text"
-        className="input"
-        placeholder="Search..."
-      />
-      <span className="icon is-left">
-        <i className="fas fa-magnifying-glass" />
-      </span>
-
-      <span className="icon is-right" style={{ pointerEvents: 'all' }}>
-        {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-        <button
-          data-cy="clearSearchButton"
-          type="button"
-          className="delete"
+import { ChangeEvent, Dispatch, SetStateAction } from 'react';
+import { TodosFilterType } from '../../types/TodosFilterType';
+
+type Props = {
+  changeFilterBy: Dispatch<SetStateAction<TodosFilterType>>;
+  query: string;
+  changeQuery: Dispatch<SetStateAction<string>>;
+  onClearField: () => void;
+};
+
+export const TodoFilter: React.FC<Props> = (props) => {
+  const {
+    changeFilterBy,
+    query,
+    changeQuery,
+    onClearField,
+  } = props;
+
+  const handleTextChange = (event: ChangeEvent<HTMLInputElement>) => {
+    changeQuery(event.target.value.trim());
+  };
+
+  const handleChangeFilterType = (event: ChangeEvent<HTMLSelectElement>) => {
+    changeFilterBy(event.target.value as TodosFilterType);
+  };
+
+  return (
+    <form className="field has-addons">
+      <p className="control">
+        <span className="select">
+          <select
+            onChange={handleChangeFilterType}
+            data-cy="statusSelect"
+          >
+            <option value="all">
+              {TodosFilterType.Default}
+            </option>
+
+            <option value="active">
+              {TodosFilterType.Active}
+            </option>
+
+            <option value="completed">
+              {TodosFilterType.Completed}
+            </option>
```

Comment:
I believe you can map through enum and write it in shorter way ;)
Also, don't forget to use enum for setting values inside of your options